src/user.rs

```diff
@@ -424,25 +408,165 @@ async fn on_room_message(
+
+    log::info!("Attempt to create a client");
+
+    let timeout = Duration::from_secs(30);
+
+    let request_config = if retry_enabled {
+        RequestConfig::new().retry_timeout(timeout)
+    } else {
+        RequestConfig::new().disable_retry().timeout(timeout)
+    };
+
+    let client = Client::builder()
+        .request_config(request_config)
+        .homeserver_url(homeserver)
+        .build()
+        .await;
+
+    match client {
+        Ok(client_value) => {
+            log::info!("New client created {}", instant.elapsed().as_millis());
+
+            Some(client_value)
+        }
+        Err(_) => {
+            println!("got request config error {}", client.err().unwrap());
+            log::info!("Failed to create client");
+            None
+        }
     }
 }
 
+struct UserParams {
+    server: String,
+    progress_bar: ProgressBar,
+    i: i64,
+    retry_attempts: usize,
+    client: Client,
+    tx: Sender<Event>,
+    respect_login_well_known: bool,
+    retry_enabled: bool,
+}
+
+fn create_user(user_params: UserParams) -> impl futures::Future<Output = Option<User<Registered>>> {
+    let client_arc = Arc::new(tokio::sync::Mutex::new(user_params.client));
+    let progress_bar = user_params.progress_bar.clone();
+
+    async move {
+        let id = format!("user_{}", user_params.i);
+
+        for _ in 0..user_params.retry_attempts {
+            let user = User::new_with_client(
+                &id,
+                &user_params.server,
+                user_params.tx.clone(),
+                client_arc.clone(),
+                user_params.retry_enabled,
+                user_params.respect_login_well_known,
+            )
+            .await;
+
+            if let Some(mut user) = user {
+                if let Some(user) = user.register().await {
+                    log::info!("User {} is now registered", user.id());
+                    progress_bar.inc(1);
+                    return Some(user);
+                }
+            }
+        }
+
+        //TODO!: This should panic or abort somehow after exhausting all retries of creating the user
+        log::info!("Couldn't init a user");
+        progress_bar.inc(1);
+        None
+    }
+}
+
+pub async fn create_desired_users(config: &Configuration, tx: Sender<Event>) {
+    let users_to_create = config.user_count;
+
+    let mut current_users = load_users(config.users_filename.clone());
+
+    let mut users = vec![];
+    let progress_bar = create_progress_bar(
+        "Init users".to_string(),
+        users_to_create.try_into().unwrap(),
+    );
+    progress_bar.tick();
+
+    let homeserver_url = config.homeserver_url.clone();
+
+    let client = get_client(&homeserver_url, config.retry_request_config)
+        .await
+        .unwrap();
+
+    let _actual_users = current_users.get_available_users(homeserver_url.clone());
+
+    let actual_user_count = match _actual_users {
+        Some(val) => val.available,
+        None => 0,
+    };
```

Comment:
Prefer using map + unwrap_or here:
```suggestion
    let actual_user_count = _actual_users.map(|users| users.available).unwrap_or(0);
```